ui/src/vayla-design-lib/datepicker/datepicker.tsx

```diff
@@ -55,49 +120,57 @@ function getHeaderElement({
-    );
-});
+const DATE_PICKER_POPUP_LEFT_PAD_PX = 14;
 
-DatePickerInput.displayName = 'DatePickerInput';
+export const DatePicker: React.FC<DatePickerProps> = ({ onChange, value, wide, ...props }) => {
+    const [open, setOpen] = React.useState(false);
+    const ref = React.useRef<HTMLInputElement>(null);
+    const className = createClassName(styles['datepicker'], wide && styles['datepicker--wide']);
+    const [date, setDate] = React.useState<Date | undefined>(value);
 
-const DATE_PICKER_POPUP_LEFT_PAD_PX = 14;
+    React.useEffect(() => {
+        onChange(date);
+    }, [date]);
 
-export const DatePicker: React.FC<DatePickerProps> = ({ onChange, value, ...props }) => {
     return (
-        <div className={'datepicker'}>
-            <ReactDatePicker
-                renderCustomHeader={getHeaderElement}
-                dateFormat="dd.MM.yyyy"
-                locale={fi}
-                selected={value}
-                onChange={(date) => onChange && onChange(date ?? undefined)}
-                calendarStartDay={1}
-                showWeekNumbers
-                popperModifiers={[
-                    {
-                        name: 'offset',
-                        fn: (state) => ({
-                            ...state,
-                            x: state.x + DATE_PICKER_POPUP_LEFT_PAD_PX,
-                        }),
-                    },
-                ]}
-                customInput={<DatePickerInput {...props} />}
+        <div className={className}>
+            <DatePickerInput
+                openDatePicker={() => setOpen(true)}
+                date={date}
+                setDate={setDate}
+                wide={wide}
+                ref={ref}
+                {...props}
             />
+            {open && (
+                <CloseableModal
+                    onClickOutside={() => setOpen(false)}
+                    offsetY={ref.current?.getBoundingClientRect().height ?? 0}
+                    positionRef={ref}
+                    className={styles['datepicker__popup-container']}>
+                    <ReactDatePicker
+                        renderCustomHeader={getHeaderElement}
+                        locale={fi}
+                        selected={value}
+                        onChange={(date) => {
+                            setDate(date ?? undefined);
+                            setOpen(false);
+                        }}
+                        calendarStartDay={1}
+                        showWeekNumbers
+                        inline
+                        popperModifiers={[
+                            {
+                                name: 'offset',
+                                fn: (state) => ({
+                                    ...state,
+                                    x: state.x + DATE_PICKER_POPUP_LEFT_PAD_PX,
+                                }),
+                            },
+                        ]}
+                    />
+                </CloseableModal>
```

Comment:
🥩 

`DatePicker`:in popup piirtyy nyt omaksi leijuvaksi elementikseen `<body>`:n lapseksi, eikä se siten ole esim. rajoitettu piirtymään parenttinsa tilan sisälle